pkg/codegen/testing/test/testdata/simple-schema-pyproject/python/pyproject.toml

```diff
@@ -0,0 +1,21 @@
+[project]
+  name = "pulumi_example"
+  description = "This is a test package for pyproject.toml"
+  dependencies = ["parver>=0.2.1", "pulumi", "semver>=2.8.1"]
```

Comment:
Is there a reason we don't set a minimum Pulumi version? I'm thinking of the regression introduced in #10346.

Could we make this require `pulumi>=` some baseline that we can periodically increment?

Reply:
Other than BWC, I don't think there's a compelling reason not to. I think it will require a little more discussion before we can determine the cadence for bumping this value, and the initial value itself.

Reply:
@AaronFriel, could you give me more information about the considerations for selecting a minimum version, or when to bump the minimum version, and I can open an issue? Or you can open the issue if you feel. NBD either way! :)